src/mountscript.sh

```diff
@@ -57,23 +57,25 @@ get_host_from_share()
         return 1
     fi
 
-    # Split host by "."
-    IFS=. read -r -a hostparts <<< "$host"
+    if [ "$nfs_vers" == "3" ]; then
+        # Split host by "."
+        IFS=. read -r -a hostparts <<< "$host"
 
-    account="${hostparts[0]}"
+        account="${hostparts[0]}"
 
-    for part in "${hostparts[@]}"; do
-       if [[ "$part" == "file" || "$part" == "blob" ]]; then
-            hostprefix="$part"
-            break
-        fi
-    done
+        for part in "${hostparts[@]}"; do
+        if [[ "$part" == "blob" ]]; then
+                hostprefix="$part"
+                break
+            fi
+        done
 
-    # Check if the prefix matches the expected azprefix
-    if [ "$hostprefix" != "$azprefix" ]; then
-        echo "Bad share name: ${hostshare}."
-        echo "Share must be of the form 'account.$azprefix.core.windows.net:/account/container' for vers=$nfs_vers"
-        return 1
+        # Check if the prefix matches the expected azprefix
+        if [ "$hostprefix" != "$azprefix" ]; then
+            echo "Bad share name: ${hostshare}."
+            echo "Share must be of the form 'account.$azprefix.core.windows.net:/account/container' for vers=$nfs_vers"
```

Comment:
Since we suppot oher forms of fqdn, I think we should emit a more generic error here. Also, the log at L56 should be same as this.
Also the "for vers=" is not very helpful.

```
echo "Share to be mounted must be of the form '<endpoint fqdn>:/account/container', where most common endpoint fqdn is 'account.$azprefix.core.windows.net', but can have one of the other supported forms"
```